QPM/SymLinker.Linker/Linker.cs

```diff
@@ -0,0 +1,269 @@
+        /// <param name="dest">Destination directory path</param>
+        /// <returns>
+        /// Returns true if the system is ready to perform a SymLink
+        /// </returns>
+        private bool CheckLinkReadiness(string source, string dest)
+        {
+            try
+            {
+                var destDrive = dest.Substring(0, 1)[0];
+                var fileSize = 0L;
+                var freeSpace = 0L;
+
+                // Check format
+                if (!char.IsLetter(destDrive))
+                    errors.Add(Error.DestinationNotAbsolutePath);
+
+                // Check existance
+                if (File.Exists(source))
+                {
+                    fileSize = new FileInfo(source).Length;
+                    if (fileSize < 1024) WriteWarn("File less than standard block size");
+                }
+                else
+                {
+                    errors.Add(Error.FileNotFound);
+                    fileSize = -1;
+                }
+
+                // Escape file to directory
+                if (Path.HasExtension(dest) && !Directory.Exists(dest))
+                    dest = Path.GetDirectoryName(dest) ?? string.Empty;
+
+                if (Directory.Exists(dest))
+                {
+                    freeSpace = new DriveInfo(destDrive.ToString()).AvailableFreeSpace;
+                }
+                else
+                {
+                    errors.Add(Error.DirectoryNotFound);
+                    freeSpace = 0;
+                }
+
+                // Check space
+                if (freeSpace <= fileSize)
+                {
+                    errors.Add(Error.FreeSpaceUnavailable);
+                }
+            }
+            catch (Exception ex)
+            {
+                errors.Add(Error.Exception);
+            }
+
+            return (errors.Count == 0);
+        }
+
+        /// <summary>
+        /// Resolves all errors encountered to their readable strings
+        /// </summary>
+        /// <returns>
+        /// Concatenation of all errors encountered
+        /// </returns>
+        private string ResolveError()
+        {
+            var errorBuilder = new StringBuilder();
+            var errorCount = errors.Count;
+            errorBuilder.Append(errorCount);
+            errorBuilder.Append(" error/s found: ");
+
+            for (int i = 0; i < errors.Count; ++i)
+            {
+                var error = errors[i];
+                errorBuilder.Append(ResolveError(error));
+                errorBuilder.Append(". ");
+            }
+
+            return errorBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Resolves an individual error to its readable string
+        /// </summary>
+        /// <param name="errorCode">Error code as found in <see cref="Error"/></param>
+        /// <returns>
+        /// The readable string for <paramref name="errorCode"/>
+        /// </returns>
+        private string ResolveError(Error errorCode)
```

Comment:
Gotta say, I'm not exactly a fan of this type of error handling- where information is essentially lost in conversion, but it is what it is, so whatever.